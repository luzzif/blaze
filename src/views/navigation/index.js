import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, CardContent, withStyles, Grid } from "@material-ui/core";
import { styles } from "./styles";
import { compose } from "redux";
import { ForwardButton } from "../../components/forward-button";
import { BackButton } from "../../components/back-button";
import { SearchBar } from "../../components/search-bar";
import { SettingsButton } from "../../components/settings-button";
import { RefreshButton } from "../../components/refresh-button";

class BasicNavigation extends Component {
    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {
        const { classes } = this.props;

        return (
            <Card classes={{ root: classes.root }} square elevation={4}>
                <CardContent classes={{ root: classes.navigationBarContent }}>
                    <BackButton />
                    <ForwardButton />
                    <RefreshButton />
                    <SearchBar />
                    <SettingsButton />
                </CardContent>
            </Card>
        );
    }
}

const enhanceComponent = compose(withStyles(styles));

export const Navigation = enhanceComponent(BasicNavigation);
