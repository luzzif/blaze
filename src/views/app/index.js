import React, { Component, Fragment } from "react";
import { Navigation } from "../navigation";
import { WebView } from "../../components/web-view";

export class App extends Component {
    render() {
        return (
            <Fragment>
                <Navigation />
                <WebView />
            </Fragment>
        );
    }
}
