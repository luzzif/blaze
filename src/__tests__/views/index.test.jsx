import React from "react";
import { App } from "./../../views/app";
import TestRenderer from "react-test-renderer";

describe("app", () => {
    let testInstance = null;

    beforeAll(() => {
        testInstance = TestRenderer.create(<App />);
    });

    it("renders without crashing", () => {
        expect(testInstance).toMatchSnapshot();
        expect(testInstance.root).not.toBeNull();
    });
});
