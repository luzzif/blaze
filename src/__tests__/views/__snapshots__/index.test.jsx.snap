exports[`app renders without crashing 1`] = `
Array [
  <div
    className="MuiPaper-root-4 MuiPaper-elevation4-10 MuiCard-root-3 BasicNavigation-root-1">
    <div
        className="MuiCardContent-root-31 BasicNavigation-navigationBarContent-2">
        <button
            className="MuiButtonBase-root-39 MuiIconButton-root-33 BasicBackButton-root-32"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button">
            <span
                className="MuiIconButton-label-38">
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-42"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24">
                    <path
                        d="M0 0h24v24H0z"
                        fill="none" />
                    <path
                        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
                </svg>
            </span>
            <span
                className="MuiTouchRipple-root-49" />
        </button>
        <button
            className="MuiButtonBase-root-39 MuiIconButton-root-33 BasicForwardButton-root-56"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button">
            <span
                className="MuiIconButton-label-38">
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-42"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24">
                    <path
                        d="M0 0h24v24H0z"
                        fill="none" />
                    <path
                        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z" />
                </svg>
            </span>
            <span
                className="MuiTouchRipple-root-49" />
        </button>
        <button
            className="MuiButtonBase-root-39 MuiIconButton-root-33 BasicRefreshButton-root-57"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button">
            <span
                className="MuiIconButton-label-38">
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-42"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24">
                    <path
                        d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
                    <path
                        d="M0 0h24v24H0z"
                        fill="none" />
                </svg>
            </span>
            <span
                className="MuiTouchRipple-root-49" />
        </button>
        <div
            className="MuiGrid-container-58 MuiGrid-justify-xs-center-75">
            <div
                className="MuiGrid-item-59 MuiGrid-grid-md-6-120">
                <div
                    className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 BasicSearchBar-root-155"
                    style={
                        Object {
                            "width": "auto",
                          }
                    }>
                    <div
                        className="MuiCardContent-root-31 BasicSearchBar-content-156">
                        <div
                            className="MuiGrid-container-58 MuiGrid-align-items-xs-center-66">
                            <div
                                className="MuiGrid-item-59 MuiGrid-grid-md-11-125">
                                <input
                                    className="BasicSearchBar-input-157"
                                    type="text" />
                            </div>
                            <div
                                className="MuiGrid-item-59 MuiGrid-grid-md-1-115">
                                <div
                                    className="MuiGrid-container-58 MuiGrid-justify-xs-flex-end-76">
                                    <div
                                        className="MuiGrid-item-59">
                                        <button
                                            className="MuiButtonBase-root-39 MuiIconButton-root-33 BasicFavoriteButton-root-158"
                                            disabled={false}
                                            onBlur={[Function]}
                                            onFocus={[Function]}
                                            onKeyDown={[Function]}
                                            onKeyUp={[Function]}
                                            onMouseDown={[Function]}
                                            onMouseLeave={[Function]}
                                            onMouseUp={[Function]}
                                            onTouchEnd={[Function]}
                                            onTouchMove={[Function]}
                                            onTouchStart={[Function]}
                                            tabIndex="0"
                                            type="button">
                                            <span
                                                className="MuiIconButton-label-38">
                                                <svg
                                                    aria-hidden="true"
                                                    className="MuiSvgIcon-root-42"
                                                    color={undefined}
                                                    focusable="false"
                                                    viewBox="0 0 24 24">
                                                    <path
                                                        d="M0 0h24v24H0z"
                                                        fill="none" />
                                                    <path
                                                        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z" />
                                                </svg>
                                            </span>
                                            <span
                                                className="MuiTouchRipple-root-49" />
                                        </button>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <button
            className="MuiButtonBase-root-39 MuiIconButton-root-33 BasicSettingsButton-root-159"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button">
            <span
                className="MuiIconButton-label-38">
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-42"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24">
                    <path
                        d="M0 0h24v24H0z"
                        fill="none" />
                    <path
                        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
                </svg>
            </span>
            <span
                className="MuiTouchRipple-root-49" />
        </button>
    </div>
</div>,
  <webview
    className="BasicWebView-root-160"
    src="http://www.github.com" />,
]
`;
